### PR TITLE
Create autodocumentation of API for GPFflow.

### PR DESCRIPTION
**Pull request by condnsdmatters**
The script generate_rst.py creates a folder `gpflow` which contains all
the rst for the autogeneration using sphinx. Also added is type checking
to the documentation and css and logo for the docs.

### CODE REVIEW

**Comment by condnsdmatters** [comment]
This is a pull request for the changes to the documentation. It is a simple script that traverses the gpflow module and creates rst for the publically accessible items on the module. It also adds some css and bits like that to the conf.py

Before merging we add this to the circle CI and make sure either it pushes to read the docs in some way, or that it publishes to github pages.

**Comment by awav** [comment]
@condnsdmatters , have you checked that it works on _readthedocs_?

**Comment by condnsdmatters** [comment]
> @condnsdmatters , have you checked that it works on _readthedocs_?

I haven't logged on to read the docs, because I don't have your login details, but it works as a hosted site (as I've shared elsewhere). The way the works is that the script generates `.rst` files,  by traversing the `gpflow` module, and then sphinx can use these autogen'd files to make a static html site, with tthe right theme. 

My remaining thoughts for this PR is that 
1) if we stick with read the docs, we need to find a way to keep our autogen'd rst auto-updating (perhaps by regenerating documentation as a final step in the pipeline, though I dont like this as it involves the commiting autogenerated files to the repo)
2) we move to github pages, which will store all our documenation in either this repo or another one. preferably this one using this [method](https://circleci.com/blog/deploying-documentation-to-github-pages-with-continuous-integration/). The advantage of this is we control everything, don;t rely on read the docs at all and can shut it down. Also, we can then set up a CNAME redirect from https://gpflow.org/documentation or https://docs.gpflow.org/ smth like that, so the whole thing is coherent

**Comment by awav** [comment]
Hello @condnsdmatters , I apologize for closing your PR. It was closed automatically when I moved `develop-2.0` to the `develop`. Could you re-open the PR? Thanks!

**Comment by condnsdmatters** [comment]
@awav This is finally visible on read the docs at https://gpflow.readthedocs.io/en/eric-documentation_patches/gpflow/covariances/index.html 

Its _fully_ ready for review.

File: setup.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 # Dependencies of GPflow
 requirements = [
+    'gast==0.2.2',
```
Comment (awav) [inline setup.py]: `gast` is a third party library, and this dependency should be resolved by `tfp` probably. Why did you put it here with this version?

**Pull request merged**